From 4e557d7a7fc766d45e5235fc3ad0d2050ca0a0ef Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Thu, 28 Jul 2016 13:41:33 -0700
Subject: [PATCH] Squashed commit of the following:

commit f983c8ea3f43dfdfb43ebfecb8941901f221c378
Author: Rakesh Pallerla <rakesh@codeaurora.org>
Date:   Thu Feb 4 14:46:43 2016 +0530

    Do not go Fullscreen in StkInputAcvtivity.

    In landscape mode if user goes to fullscreen password is not
    masked and is visible. To avoid this do not go to full screen
    in StkInputActivity.

    Change-Id: I70d08c00ca1c4f9c05f7982a300ac4bf9fc8e1be
    CRs-Fixed: 945011

commit e1ec2ccfc756bfb9b68fd6c31e32ae6b9b157bb1
Author: Umashankar Godachi <umasha@codeaurora.org>
Date:   Wed Mar 30 19:41:31 2016 +0530

    Fix the notification cancel during SIM refresh.

    During SIM refresh, the idle mode text needs to
    be cleared, currently it is always cleared for
    default subscription irrespective of the slot
    on which the refresh is received.

    Fix: In StkAppService, while handling the refresh
    get the notification id for the respective slot and
    do the cancel.

    Change-Id: I861656188842809fb248d94710f6407f210829c9
    CRs-Fixed: 988831

commit dec74008a61e571cb9c4aaabdc7b66b1d886515f
Author: Ravindra <c_rthat@codeaurora.org>
Date:   Fri Feb 19 17:10:48 2016 +0530

    MSIM: Add MSIM support for language selection event.

    Currently for language selection event received from
    SIM 2 terminal response is not sent from ME.

    Fix: Check if set up event response is needed for all
    the available slots and send the terminal response for
    the slot on which the set up event is received.

    Change-Id: I0699a537737ea658802e1361eb9863de0d35f31f
    CRs-Fixed: 973287

commit 6dd1eb93b568b4dcfe6741e6392cc492487b03c2
Author: Wileen Chiu <wileenc@codeaurora.org>
Date:   Mon Mar 14 12:18:31 2016 -0700

    Send slotId in message for OP_STOP_TONE

    - unable to send the response to the
    correct sim card
    - Set the slotId as arg2 in the message

    Change-Id: I31143b35af5d19077eb7224736aef9c664016246
    CRs-Fixed: 989267

commit 7757426e42015e8e0111000f1fbdf77ccbfeea05
Author: Paramananda Pradhan <parama@codeaurora.org>
Date:   Wed Jan 14 18:09:22 2015 +0530

    Add intent to send STK app name

    As part of this code change, a new broadcast will
    be added to send to the Launcher to display the
    operator name. The changes are under a config file
    and will be enabled in operator specific config.

    Change-Id: I336171de2ee07f0cf50483975c31ec456fd82dfd
    CRs-Fixed: 776336

Change-Id: I4e863a01aa94ac12cd57842a1fcf9ee5743b9921
---
 res/values/config.xml                     | 33 +++++++++++++++++++++++++++++++
 src/com/android/stk/StkAppService.java    | 21 ++++++++++++++++++--
 src/com/android/stk/StkInputActivity.java |  2 ++
 3 files changed, 54 insertions(+), 2 deletions(-)
 create mode 100644 res/values/config.xml

diff --git a/res/values/config.xml b/res/values/config.xml
new file mode 100644
index 0000000..a457707
--- /dev/null
+++ b/res/values/config.xml
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright (c) 2015, The Linux Foundation. All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above
+      copyright notice, this list of conditions and the following
+      disclaimer in the documentation and/or other materials provided
+      with the distribution.
+    * Neither the name of The Linux Foundation nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED
+WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT
+ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS
+BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <bool name="send_stk_title" translatable="false">false</bool>
+</resources>
diff --git a/src/com/android/stk/StkAppService.java b/src/com/android/stk/StkAppService.java
index f1a4afe..6e2e872 100755
--- a/src/com/android/stk/StkAppService.java
+++ b/src/com/android/stk/StkAppService.java
@@ -615,7 +615,11 @@ public class StkAppService extends Service implements Runnable {
                 break;
             case OP_LOCALE_CHANGED:
                 CatLog.d(this, "Locale Changed");
-                checkForSetupEvent(LANGUAGE_SELECTION_EVENT,(Bundle) msg.obj, slotId);
+                for (int slot = 0; slot < mSimCount; slot++) {
+                    if (mStkContext[slot] != null) {
+                        checkForSetupEvent(LANGUAGE_SELECTION_EVENT,(Bundle) msg.obj, slot);
+                    }
+                }
                 break;
             case OP_ALPHA_NOTIFY:
                 handleAlphaNotify((Bundle) msg.obj);
@@ -937,6 +941,18 @@ public class StkAppService extends Service implements Runnable {
                 }
             } else {
                 CatLog.d(LOG_TAG, "install App");
+                CatLog.d(LOG_TAG, "getResources().getBoolean(R.bool.send_stk_title ="
+                        + (getResources().getBoolean(R.bool.send_stk_title)));
+                if ((getResources().getBoolean(R.bool.send_stk_title))) {
+                    String stkTitle = mStkContext[slotId].mCurrentMenu.title;
+                    CatLog.d(LOG_TAG ,"mCurrentMenu.title =" + stkTitle);
+                    //Send intent with the extra to launcher
+                    final String STK_INTENT =
+                            "org.codeaurora.carrier.ACTION_TELEPHONY_SEND_STK_TITLE";
+                    Intent intent = new Intent(STK_INTENT);
+                    intent.putExtra("StkTitle", stkTitle);
+                    sendStickyBroadcast(intent);
+                }
                 StkAppInstaller.install(mContext);
             }
             if (mStkContext[slotId].mMenuIsVisible) {
@@ -978,7 +994,7 @@ public class StkAppService extends Service implements Runnable {
             launchEventMessage(slotId);
             // Idle mode text needs to be cleared for init or reset modes of refresh
             if (cmdMsg.isRefreshResetOrInit()) {
-                mNotificationManager.cancel(STK_NOTIFICATION_ID);
+                mNotificationManager.cancel(getNotificationId(slotId));
                 mStkContext[slotId].mIdleModeTextCmd = null;
                 CatLog.d(this, "Clean idle mode text due to refresh");
             }
@@ -1713,6 +1729,7 @@ public class StkAppService extends Service implements Runnable {
 
         Message msg = mServiceHandler.obtainMessage();
         msg.arg1 = OP_STOP_TONE;
+        msg.arg2 = slotId;
         msg.obj = (Integer)(showUserInfo ? 1 : 0);
         msg.what = STOP_TONE_WHAT;
         mServiceHandler.sendMessageDelayed(msg, timeout);
diff --git a/src/com/android/stk/StkInputActivity.java b/src/com/android/stk/StkInputActivity.java
index 021c832..e8d6a4d 100644
--- a/src/com/android/stk/StkInputActivity.java
+++ b/src/com/android/stk/StkInputActivity.java
@@ -28,6 +28,7 @@ import android.text.InputFilter;
 import android.text.InputType;
 import android.text.TextWatcher;
 import android.text.method.PasswordTransformationMethod;
+import android.view.inputmethod.EditorInfo;
 import android.view.KeyEvent;
 import android.view.MenuItem;
 import android.view.View;
@@ -394,6 +395,7 @@ public class StkInputActivity extends Activity implements View.OnClickListener,
                 mTextIn.setTransformationMethod(PasswordTransformationMethod
                         .getInstance());
             }
+            mTextIn.setImeOptions(EditorInfo.IME_FLAG_NO_FULLSCREEN);
             // Set default text if present.
             if (mStkInput.defaultText != null) {
                 mTextIn.setText(mStkInput.defaultText);
-- 
2.7.4

